src/instruction.rs

```diff
@@ -0,0 +1,422 @@
+    Instruction{name: "SBC", opcode: 0xF5, mode: Mode::ZeroPageX,   len: 2},
+    Instruction{name: "SBC", opcode: 0xED, mode: Mode::Absolute,    len: 3},
+    Instruction{name: "SBC", opcode: 0xFD, mode: Mode::AbsoluteX,   len: 3},
+    Instruction{name: "SBC", opcode: 0xF9, mode: Mode::AbsoluteY,   len: 3},
+    Instruction{name: "SBC", opcode: 0xE1, mode: Mode::IndirectX,   len: 2},
+    Instruction{name: "SBC", opcode: 0xF1, mode: Mode::IndirectY,   len: 2}
+];
+
+const STA_SET: [Instruction; 7] = [
+    Instruction{name: "STA", opcode: 0x85, mode: Mode::ZeroPage,    len: 2},
+    Instruction{name: "STA", opcode: 0x95, mode: Mode::ZeroPageX,   len: 2},
+    Instruction{name: "STA", opcode: 0x8D, mode: Mode::Absolute,    len: 3},
+    Instruction{name: "STA", opcode: 0x9D, mode: Mode::AbsoluteX,   len: 3},
+    Instruction{name: "STA", opcode: 0x99, mode: Mode::AbsoluteY,   len: 3},
+    Instruction{name: "STA", opcode: 0x81, mode: Mode::IndirectX,   len: 2},
+    Instruction{name: "STA", opcode: 0x91, mode: Mode::IndirectY,   len: 2}
+];
+
+const STACK_SET: [Instruction; 6] = [
+    Instruction{name: "TXS", opcode: 0x9A, mode: Mode::Implicit,    len: 1},
+    Instruction{name: "TSX", opcode: 0xBA, mode: Mode::Implicit,    len: 1},
+    Instruction{name: "PHA", opcode: 0x48, mode: Mode::Implicit,    len: 1},
+    Instruction{name: "PLA", opcode: 0x68, mode: Mode::Implicit,    len: 1},
+    Instruction{name: "PHP", opcode: 0x08, mode: Mode::Implicit,    len: 1},
+    Instruction{name: "PLP", opcode: 0x28, mode: Mode::Implicit,    len: 1}
+];
+
+const STX_SET: [Instruction; 3] = [
+    Instruction{name: "STX", opcode: 0x86, mode: Mode::ZeroPage,    len: 2}, // 0x86
+    Instruction{name: "STX", opcode: 0x96, mode: Mode::ZeroPageY,   len: 2},
+    Instruction{name: "STX", opcode: 0x8E, mode: Mode::Absolute,    len: 3},
+];
+
+const STY_SET: [Instruction; 3] = [
+    Instruction{name: "STY", opcode: 0x84, mode: Mode::ZeroPage,    len: 2},
+    Instruction{name: "STY", opcode: 0x94, mode: Mode::ZeroPageX,   len: 2},
+    Instruction{name: "STY", opcode: 0x8C, mode: Mode::Absolute,    len: 3},
+];
+
+const BRK: Instruction = Instruction{name: "BRK", opcode: 0x00, mode: Mode::Implicit, len: 1};
+const JSR: Instruction = Instruction{name: "JSR", opcode: 0x20, mode: Mode::Absolute, len: 3};
+const NOP: Instruction = Instruction{name: "NOP", opcode: 0xEA, mode: Mode::Implicit, len: 1};
+const RTI: Instruction = Instruction{name: "RTI", opcode: 0x40, mode: Mode::Implicit, len: 1};
+const RTS: Instruction = Instruction{name: "RTS", opcode: 0x60, mode: Mode::Implicit, len: 1};
+
+lazy_static! {
+    static ref INSTRUCTIONS: Vec<&'static Instruction> = {
+        let mut v = Vec::new();
+        v.push(&BRK);
+        v.push(&JSR);
+        v.push(&NOP);
+        v.push(&RTI);
+        v.push(&RTS);
+        for instruction in ADC_SET.iter()
+        {
+            v.push(instruction);
+        }
```

Comment:
i think that its a lot of for loops :D https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice

Reply:
I'll change it